presentation/src/main/java/com/example/user/test/presentation/screen/splash/SplashActivity.kt

```diff
@@ -35,11 +35,7 @@ class SplashActivity : BaseActivity(), SplashView {
     override fun onCreate(savedInstanceState: Bundle?) {
         MarvelApplication.appComponent.inject(this)
         super.onCreate(savedInstanceState)
-    }
-
-    override fun onResume() {
-        super.onResume()
 
-        presenter.goToMainActivity()
+        presenter.onFirstCreateCalled()
```

Comment:
Why do you call it in `onResume`? Call it into `onCreate` and not for restoration case.